File: CodingSamples/DotNet/Fundamentals/Inheritance/InterfaceTypeTest/DemoApp2/Auditor.cs
using Taxation;

class Auditor : IDisposable
{
    public Auditor()
    {
        Console.WriteLine("Auditor Log [{0}] - starting audit session", DateTime.Now);
    }

    public void Audit(string id, ITaxPayer employee)
    {
        Console.WriteLine("Auditing {0}...", id);
        if(id.Length < 4)
            throw new ArgumentException("Invalid ID");
        decimal payment = employee.IncomeTax() + 500;
        Console.WriteLine("Total tax payment: {0:0.00}", payment);
    }

    public void Dispose()
    {
        Console.WriteLine("Auditor Log [{0}] - ending audit session", DateTime.Now);
    }
}

File: CodingSamples/Java/Fundamentals/Fundamentals/Inheritance/InterfaceTest/demo2/Auditor.java
import java.util.Date;

import taxation.TaxPayer;

class Auditor implements AutoCloseable {

    public Auditor() {
        System.out.printf("Auditor Log [%s] - starting audit session.%n", new Date());
    }

    public void audit(String id, TaxPayer employee) {
        System.out.printf("Auditing %s...%n", id);
        if(id.length() < 4)
            throw new IllegalArgumentException("Invalid ID");
        double payment = employee.incomeTax() + 500;
        System.out.printf("Total tax payment: %.2f%n", payment);
    }

    public void close() {
        System.out.printf("Auditor Log [%s] - ending audit session.%n", new Date());
    }
}
